src/main/java/com/meilisearch/sdk/Client.java

```diff
@@ -68,6 +68,37 @@ public Index createIndex(String uid, String primaryKey) throws Exception {
 		return meiliSearchIndexList;
 	}
 
+	
+
+	/**
+	 * Create a local reference to an index identified by `uid`, without doing an HTTP call.
+	 * Calling this method doesn't create an index by itself, but grants access to all the other methods in the Index class.
+	 *
+	 * @param uid Unique identifier of the index
+	 * @return Index instance
+	 * @throws Exception If an error occurs
+	 */
+	public Index index(String uid) throws Exception {
+		return this.index(uid, null);
+	}
+
+	/**
+	 * Create a local reference to an index identified by `uid`, without doing an HTTP call.
+	 * Calling this method doesn't create an index by itself, but grants access to all the other methods in the Index class.
+	 *
+	 * @param uid Unique identifier of the index
+	 * @param primaryKey The primary key of the documents in that index
+	 * @return Index instance
+	 * @throws Exception If an error occurs
+	 */
+	public Index index(String uid, String primaryKey) throws Exception {
```

Comment:
We decide the `index` method should not take a primary key as parameter since it's only a method to define a scope.
We don't use it to do `client.index(uid, primary_key).create()` or `client.index(uid, primary_key).add_documents(...)`, for the moment -> it would mean take the primary key (in RAM) into account and this primary could lead to error/confusion since it could be different from the remote one. That's the decision we took (after a long debate): stay simple for the moment by not including the primary key into `index` method.